File: Summer_Winter_Coding(~2018)/소수 만들기-p12977/소수 만들기.py
// https://school.programmers.co.kr/learn/courses/30/lessons/12977

def solution(nums):
    answer = 0
    combinations = []

    for i in range(len(nums) -2):
        for j in range(i+1, len(nums) -1):
            for k in range(j+1, len(nums)):
                combinations.append([nums[i], nums[j], nums[k]])

    for combi in combinations:
        sum = combi[0] + combi[1] + combi[2]
        flag = True;

        for i in range(2, sum):
            if ( sum % i == 0):
                flag = False

        if (flag):
            print(sum)
            answer += 1

    return answer

File: Summer_Winter_Coding(~2018)/소수 만들기-p12977/소수 만들기.js
// https://school.programmers.co.kr/learn/courses/30/lessons/12977

function solution(nums) {
    let answer = 0;
    const CNT = 3;
    const combinations = [];

    for (i = 0 ; i <= nums.length-CNT ; i++) {
        for(j = i+1 ; j < nums.length-1 ; j++) {
            for(k=j+1; k < nums.length ; k++) {
                const combi = [];
                combi.push(nums[i], nums[j], nums[k]);
                combinations.push(combi);
            }
        }
    }

    combinations.forEach((combi) => {
        const sum = combi[0] + combi[1] + combi[2];
        let flag = true;

        for(i = 2; i < sum; i++) {
            if (sum % i === 0) {
                flag = false;
            }
        }

        if (flag) answer++;
    });

    return answer;
}